scripts/pylib/pytest-twister-harness/src/twister_harness/fixtures.py

```diff
@@ -58,7 +59,13 @@ def shell(dut: DeviceAdapter) -> Shell:
     """Return ready to use shell interface"""
     shell = Shell(dut, timeout=20.0)
     logger.info('Wait for prompt')
-    assert shell.wait_for_prompt()
+    if not shell.wait_for_prompt():
+        pytest.fail('Prompt not found')
+    if dut.device_config.type == 'hardware':
+        # after booting up the device, there might appear additional logs
+        # after first prompt, so we need to wait and clear the buffer
+        time.sleep(0.5)
```

Comment:
maybe we can change to just wait for shell prompt '#' to appear.

Reply:
Thanks for suggestion. I also though about that solution to run `shell.wait_for_prompt()` once again, but I wanted to avoid doing those extra operations (sending new line to device, logging new messages, etc.). In cases I've tested on nRF boards it was enough just to clear that buffer. If more action will be required to execute, than it can be run from test script, or from additional fixture added for a specific platform / scenario.